File: TemplateMapper/TemplateMapper/Models/TaskDuration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Models
{
    public class TaskDuration
    {
        public long TaskDurationId { get; set; }
        public string Name { get; set; }
        public TimeSpan Duration { get; set; }
        public DateTime TimeStamp { get; set; }
        public long DrawingId { get; set; }

        public virtual Drawing Drawing { get; set; }

        public TaskDuration()
        { }

        public TaskDuration(string name, DateTime previous)
        {
            Name = name;
            TimeStamp = DateTime.Now;
            Duration = TimeStamp - previous;
        }
    }
}

File: DataSummitSolution/DataSummitUI/src/app/efModels/taskDuration.ts
import { Timestamp } from "rxjs/internal/operators/timestamp";
import { Drawing } from "../drawings/models/drawing";
import { TimeSpan } from "../shared/models/timespan";

export class TaskDuration {

	TaskDurationId: number;
	Name: string;
	TimeStamp: Date | string;
	DrawingId: number;
	Duration: TimeSpan;
	Drawing: Drawing;

	constructor(taskDurationId?: number, name?: string, timeStamp?: Date | string,
		drawingId?: number, duration?: TimeSpan, drawing?: Drawing)
	{
		this.TaskDurationId = taskDurationId || 0;
		this.Name = name || "";
		this.TimeStamp = timeStamp || new Date(Date.now());
		this.DrawingId = drawingId || 0;
		this.Duration = duration;
		this.Drawing = drawing;
	}
}
